File: app/src/main/java/com/example/amyfunk/ad340app/TrafficCam.java
package com.example.amyfunk.ad340app;

public class TrafficCam {
    private String description;
    private String imageUrl;
    private String type;
    private double[] coords;

    public TrafficCam(String description, String imageUrl, String type, double[] coords){

        this.description = description;
        this.imageUrl = imageUrl;
        this.type = type;
        this.coords = coords;

    }
    public String getDescription(){
        return description;
    }

    public String getImageUrl(){
        return imageUrl;
    }

    public String getType(){
        return type;
    }

    public double[] getCoords() {
        return coords;
    }
}

File: app/src/main/java/com/example/amyfunk/ad340app/TrafficCam.kt
package com.example.amyfunk.ad340app

class TrafficCam(val description: String, val image: String, var type: String, var coords: DoubleArray ) {

    val baseUrl: Map<String, String> = mapOf(
            "sdot" to "http://www.seattle.gov/trafficcams/images/",
            "wsdot" to "http://images.wsdot.wa.gov/nw/"
    )

    fun imageUrl() : String {
        return baseUrl[this.type] + this.image
    }

    fun coords() : DoubleArray {
        return coords
    }

    fun getDesc() : String {
        return description
    }
}